README.md
# Wikiwars Website

[![Netlify Status](https://api.netlify.com/api/v1/badges/56e73fb9-520b-4749-bc91-8d72b9da4e6f/deploy-status)](https://app.netlify.com/sites/wikiwarsgg/deploys)

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## 0.1.0 (2019-07-02)


gridsome.config.js
// This is where project configuration and plugin options are located.
// Learn more: https://gridsome.org/docs/config

// Changes here require a server restart.
// To restart press CTRL + C in terminal and run `gridsome develop`

const tailwind = require('tailwindcss')
const purgecss = require('@fullhuman/postcss-purgecss')

const postcssPlugins = [
  tailwind(),
]

if (process.env.NODE_ENV === 'production') postcssPlugins.push(purgecss())

module.exports = {
  siteName: 'WikiWars',
  siteDescription: 'A simple portfolio theme for Gridsome powered by Tailwind CSS v1.0',
  siteUrl: 'https://www.wikiwars.gg',
  plugins: [
    {
      use: '@gridsome/source-filesystem',
      options: {
        path: 'blog/**/*.md',
        typeName: 'Post',
        refs: {
          tags: {
            typeName: 'Tag',
            route: 'tag/:id',
            create: true
          }
        },
        remark: {
          plugins: [
            [ 'gridsome-plugin-remark-shiki', { theme: 'Material-Theme-Palenight', skipInline: true } ]
          ]
        }
      }
    },
    {
      use: 'gridsome-plugin-rss',
      options: {
        contentTypeName: 'Post',
        feedOptions: {
          title: 'Gridsome Portfolio Starter Blog',
          feed_url: 'https://www.wikiwars.gg/rss.xml',
          site_url: 'https://www.wikiwars.gg/'
        },
        feedItemOptions: node => ({
          title: node.title,
          description: node.summary,
          url: 'https://www.wikiwars.gg' + node.path,
          author: 'YES WE CAPS',
          date: node.date
        }),
        output: {
          dir: './static',
          name: 'rss.xml'
        }
      }
    },
    {
      use: '@gridsome/plugin-sitemap',
      options: {
        cacheTime: 600000, // default
      }
    },
  ],
  transformers: {
    remark: {
      externalLinksTarget: '_blank',
      externalLinksRel: ['nofollow', 'noopener', 'noreferrer'],
      anchorClassName: 'icon icon-link',
    }
  },
  css: {
    loaderOptions: {
      postcss: {
        plugins: postcssPlugins,
      },
    },
  },
}


tailwind.config.js
module.exports = {
  theme: {
    extend: {
      spacing: {
        "80": "20rem",
        "108": "27rem",
      },
      borderWidth: {
        "14": "14px",
      },
    },
    container: {
      padding: "1rem",
    },
    colors: {
      background: {
        primary: "var(--bg-background-primary)",
        secondary: "var(--bg-background-secondary)",
        tertiary: "var(--bg-background-tertiary)",

        form: "var(--bg-background-form)",
      },

      copy: {
        primary: "var(--text-copy-primary)",
        secondary: "var(--text-copy-hover)",
      },

      "border-color": {
        primary: "var(--border-border-color-primary)",
      },

      transparent: "transparent",

      black: "#000",
      white: "#fff",
      blue: {
        100: "#ebf8ff",
        200: "#bee3f8",
        300: "#90cdf4",
        400: "#63b3ed",
        500: "#4299e1",
        600: "#3182ce",
        700: "#2b6cb0",
        800: "#2c5282",
        900: "#2a4365",
        full: "#00f",
      },
      green: {
        100: "#f0fff4",
        200: "#c6f6d5",
        300: "#9ae6b4",
        400: "#68d391",
        500: "#48bb78",
        600: "#38a169",
        700: "#2f855a",
        800: "#276749",
        900: "#22543d",
      },

      gray: {
        100: "#f7fafc",
        200: "#edf2f7",
        300: "#e2e8f0",
        400: "#cbd5e0",
        500: "#a0aec0",
        600: "#718096",
        700: "#4a5568",
        800: "#2d3748",
        900: "#1a202c",
      },
    },
    fontFamily: {
      sans: [
        "Nunito Sans",
        "Roboto",
        "-apple-system",
        "BlinkMacSystemFont",
        '"Segoe UI"',
        '"Helvetica Neue"',
        "Arial",
        '"Noto Sans"',
        "sans-serif",
        '"Apple Color Emoji"',
        '"Segoe UI Emoji"',
        '"Segoe UI Symbol"',
        '"Noto Color Emoji"',
      ],
      serif: ["Georgia", "Cambria", '"Times New Roman"', "Times", "serif"],
      mono: [
        "Menlo",
        "Monaco",
        "Consolas",
        '"Liberation Mono"',
        '"Courier New"',
        "monospace",
      ],
      bebas: ['"Bebas Neue"', '"Helvetica Neue"', "Arial"],
    },
  },
};


src/main.js
// This is the main.js file. Import global CSS and scripts here.
// The Client API can be used here. Learn more: gridsome.org/docs/client-api

import DefaultLayout from "~/layouts/Default.vue";
import VueScrollTo from "vue-scrollto";
import VueFuse from "vue-fuse";

export default function(Vue, { router, head, isClient }) {
  // Set default layout as a global component
  Vue.component("Layout", DefaultLayout);

  Vue.use(VueScrollTo, {
    duration: 500,
    easing: "ease",
  });

  Vue.use(VueFuse);

  head.meta.push(
    {
      name: "keywords",
      content: "Wikiwars,Wikipedia,Events,Paris,France",
    },
    {
      name: "description",
      content: "Des événements décalés, fun et compétitifs sur... Wikipédia!",
    },
    {
      name: "author",
      content: "Andre Madarang",
    },
    {
      name: "twitter:card",
      content: "summary",
    },
    {
      name: "twitter:title",
      content: "Wikiwars",
    },
    {
      name: "twitter:description",
      content: "Des événements décalés, fun et compétitifs sur... Wikipédia!",
    },
    {
      name: "twitter:site",
      content: "@wikiwarsgg",
    },
    {
      name: "og:title",
      content: "Wikiwars",
    },
    {
      name: "og:description",
      content: "Des événements décalés, fun et compétitifs sur... Wikipédia!",
    },
    {
      name: "og:url",
      content: "https://wikiwars.gg",
    },
    {
      name: "og:site_name",
      content: "Wikiwars",
    },
    {
      name: "og:locale",
      content: "fr_FR",
    },
    {
      name: "fb:app_id",
      content: "365853070100963",
    },
    {
      name: "og:type",
      content: "website",
    }
  );

  head.link.push({
    rel: "stylesheet",
    href: "https://fonts.googleapis.com/css?family=Nunito+Sans:400,700",
  });
}
